spec/calendar.html

```diff
@@ -260,12 +260,14 @@ <h1>GetOptionalTemporalCalendar ( _item_ )</h1>
     <emu-clause id="sec-temporal-calendarfrom" aoid="CalendarFrom">
       <h1>CalendarFrom ( _identifier_ )</h1>
       <emu-alg>
-        1. Let _from_ be ? GetMethod(%Temporal.Calendar%, *"from"*).
-        1. If _from_ is *undefined*, set _from_ to %Temporal.Calendar.from%.
-        1. Let _calendar_ be ? Call(_from_, %Temporal.Calendar%, « _identifier_ »).
-        1. If Type(_calendar_) is not Object, then
-          1. Throw a *TypeError* exception.
-        1. Return _calendar_.
+        1. If Type(_item_) is Object, then
+          1. If ? HasProperty(_item_, *"calendar"*) is *false*, return _item_.
+          1. Set _item_ to ? Get(_item_, *"calendar"*).
+          1. If Type(_item_) is Object and ? HasProperty(_item_, *"calendar"*) is *false*, return _item_.
```

Comment:
you already know HasProperty is true here, since you returned early 2 lines up

Reply:
_item_ is a different value from what it was 2 lines earlier, here. (This is for the case of `Temporal.Calendar.from(someTemporalObject)` where `someTemporalObject` has a `.calendar` property.)

Reply:
ah, i see. that's confusing; is there a reason we can't create a new spec variable?

Reply:
if it's a temporal object, why would it use `.calendar` instead of an internal slot?

Reply:
That's discussed in #925, to handle the case of a property bag standing in for a Temporal object. I think it'd be reasonable to additionally check if there was a [[Calendar]] internal slot and use that, but I don't feel comfortable reversing a decision that the champions group already had a lengthy discussion on, just as an aside in this PR. So I'd prefer that this PR preserve the existing algorithm. If you feel strongly about it, could you open another issue to discuss it?

Reply:
You're saying that this is to support a *third* type of thing (Temporal objects, strings, and Temporal-like objects)?

Yes, I'll file an issue about that.

Reply:
https://github.com/tc39/proposal-temporal/issues/1428

Reply:
I meant filing an issue about changing this particular site to check [[Calendar]], which is — as I said above — just fine with me. I just didn't want to change the algorithm _in this PR_ where the other champions might not have a chance to comment. But I guess we can handle it as part of #1428. I'll respond there separately.